Modules/_typesmodule.c

```diff
@@ -1,17 +1,52 @@
 /* _types module */
 
 #include "Python.h"
+#include "pycore_descrobject.h"   // _PyMethodWrapper_Type
 #include "pycore_namespace.h"     // _PyNamespace_Type
+#include "pycore_object.h"        // _PyNone_Type, _PyNotImplemented_Type
+#include "pycore_unionobject.h"   // _PyUnion_Type
 
 static int
 _types_exec(PyObject *m)
 {
-    if (PyModule_AddObjectRef(m, "CapsuleType", (PyObject *)&PyCapsule_Type) < 0) {
-        return -1;
-    }
-    if (PyModule_AddObjectRef(m, "SimpleNamespace", (PyObject *)&_PyNamespace_Type) < 0) {
-        return -1;
-    }
+#define EXPORT_STATIC_TYPE(NAME, TYPE)                                   \
+    do {                                                                 \
+        assert(PyUnstable_IsImmortal((PyObject *)&(TYPE)));              \
+        if (PyModule_AddObjectRef(m, (NAME), (PyObject *)&(TYPE)) < 0) { \
+            return -1;                                                   \
+        }                                                                \
+    } while (0)
+
+    EXPORT_STATIC_TYPE("AsyncGeneratorType", PyAsyncGen_Type);
+    EXPORT_STATIC_TYPE("BuiltinFunctionType", PyCFunction_Type);
+    // Same as BuiltinFunctionType
+    EXPORT_STATIC_TYPE("BuiltinMethodType", PyCFunction_Type);
```

Comment:
I think this is one of the few times where it's better to put the comment on the same line. It's not immediately clear to me that the comment is referring to `BuiltinMethodType`.
```suggestion
    EXPORT_STATIC_TYPE("BuiltinMethodType", PyCFunction_Type); // Same as BuiltinFunctionType
```

Same goes for `FunctionType` below.

Reply:
What do you think about [`0fbbf0e` (#132103)](https://github.com/python/cpython/pull/132103/commits/0fbbf0ec14d0925d2281f05d47882223acf1742b)? I can't find a nice way of putting the LambdaType comment inline without breaking line length.

A

Reply:
I don't think the line-length is that important here, but the new comment works too.

Reply:
[PEP 7](https://peps.python.org/pep-0007/#code-lay-out) says otherwise :)